wcfsetup/install/files/lib/system/template/plugin/IpSearchModifierTemplatePlugin.class.php

```diff
@@ -0,0 +1,42 @@
+<?php
+namespace wcf\system\template\plugin;
+use wcf\system\template\TemplateEngine;
+use wcf\system\WCF;
+use wcf\util\MessageUtil;
+use wcf\util\StringUtil;
+use wcf\util\Url;
+
+/**
+ * IP address search modifier plugin which links the IP address to a search engine.
+ *
+ * In case the second parameter is given, the shown IP address gets truncated based on the given amount
```

Comment:
Truncation based on a number of characters does not make sense: IPv6 addresses are way longer than IPv4 adresses and even the latter might differ greatly in length. Either they are cut too short or are cut too long.

You should null trailing bits of the IP address based on CIDR, adding a method in a `*Util` might make sense.

Reply:
This Util will also come in handy for #2691.

Reply:
> Didn't forget this one, but I'll push the code since I'm going to be away for a few days again.

Looking at this once again: You did not truncate the IP address for the search engine link, but for `textContent` of the `<a>` only. I thought this was for privacy, but apparently it is not. Why did you the second parameter?

Reply:
I just kept the existing behavior:
https://github.com/WoltLab/WCF/blob/af3d18c74b9a1abc5ec9658b48d81308d0c0e419/com.woltlab.wcf/templates/usersOnlineList.tpl#L97
I think it's there since WCF 2 where the list was splitted. Same with the user agent which gets truncated.

Reply:
I see. I'd say remove this: A full length IPv6 that is going to be seen in the wild only exceeds this by 9 characters (technically it's 45 characters total, but we convert IPv4 mapped addresses to real IPv4) and the `…` that is appended visually takes up the space of at least two characters. So there is hardly any gain. /cc @dtdesign 

I believe that the CIDR truncation is not required then either. It should rather be implemented as part of #2691.

Reply:
I don't see a point in truncating the IP either. In this particular situation, the usefulness of the ip address is strictly boolean, either I don't care about it at all or I need the exact ip address for whatever reasons. Truncating it contributes to neither of those goals, while also actively working against the user when the full ip address does matter.

Reply:
Then I'll remove the truncation.